packages/voting-contract/src/lib.rs

```diff
@@ -116,12 +116,13 @@ where
 {
     // ensure proposal exists and can be voted on
     let mut prop = proposals().load(deps.storage, proposal_id)?;
-    if prop.status != Status::Open {
+    prop.update_status(&env.block);
+
+    if prop.status == Status::Rejected {
+        return Err(ContractError::Rejected {});
```

Comment:
When I started implementing this and noticed that in-memory status won't be actually rejected, I thought that more explicit error might be more useful.
But okay, since query actually will show that as rejected I guess it's superflous.